src/sql/rewrite/ob_transform_simplify_expr.cpp

```diff
@@ -2517,4 +2526,740 @@ int ObTransformSimplifyExpr::try_remove_ora_decode(ObRawExpr *&expr,
+          and ({expr1 to exprn} \ exprk) op const_value is all false, then 
+          expr can be rewritten to lnnvl(cond1) and .. and lnnvl(condk-1) and condk and (exprk op const_value).
+          (If k == n, condk is ignored.)
+    CONSTRAINTS:
+    a. IF {expr1 to exprn} op const_value is FALSE, then
+        append(constraints, {T_OR, cond1, ..., condn} IS FALSE)
+    b. OR if {expr1 to exprn} \ exprk op const_value is FALSE, exprk is TRUE/NULL then
+        append(constraints, {T_OR, cond1, ..., condn} \ exprk IS FALSE, exprk op const_value is TRUE/NULL);
+
+ **/
+int ObTransformSimplifyExpr::convert_case_when_predicate(ObDMLStmt *stmt, 
+                                                         bool &trans_happened) {
+  int ret = OB_SUCCESS;
+  trans_happened = false;
+  bool is_happened = false;
+  bool is_scala_group_by = false;
+  if (OB_ISNULL(stmt)) {
+    ret = OB_ERR_UNEXPECTED;
+    LOG_WARN("stmt is NULL", K(*stmt));
+  } else {
+    ObStmtExprGetter visitor;
+    visitor.set_relation_scope();
+    ObSEArray<ObRawExprPointer, 8> relation_exprs;
+    is_scala_group_by = stmt->is_select_stmt() && static_cast<ObSelectStmt *>(stmt)->is_scala_group_by(); 
+    if (OB_FAIL(stmt->get_relation_exprs(relation_exprs, visitor))) {
+      LOG_WARN("failed to get relation exprs", K(ret));
+    } else {
+      for (int64_t i = 0; OB_SUCC(ret) && i < relation_exprs.count(); ++i) {
+        ObRawExpr *expr = NULL;
+        if (OB_FAIL(relation_exprs.at(i).get(expr))) {
+          LOG_WARN("failed to get expr",K(ret));
+        } else if (OB_FAIL(convert_case_when_predicate(expr, is_scala_group_by, is_happened))) {
+          LOG_WARN("failed to convert case when predicate", K(ret));
+        } else if (OB_FAIL(relation_exprs.at(i).set(expr))) {
+          LOG_WARN("failed to set expr", K(ret));
+        } else {
+          trans_happened |= is_happened;
+        }
+      }
+    }
+  }
+  return ret;
+}
+
+
+// /**
+//  * @brief convert case when predicate according to then exprs at the parent expr
+//        OR convert case when predicate according to when exprs at the case expr
+//  * @param[in][out] expr: current expr 
+//  * @param[in] is_sacla_group_by: this stmt is scala group by or not
+//  * @param[in][out] trans_happened: whether rewrite is happened
+//  */
+int ObTransformSimplifyExpr::convert_case_when_predicate(
+                                    ObRawExpr *&expr,
+                                    const bool is_scala_group_by,
+                                    bool &trans_happened) 
+{
+  int ret = OB_SUCCESS;
+  trans_happened = false;
+  bool is_happened = false;
+  if (OB_ISNULL(expr)) {
+    ret = OB_ERR_UNEXPECTED;
+    LOG_WARN("unexpected null", K(expr), K(ret));
+  } else if (IS_COMMON_COMPARISON_OP(expr->get_expr_type())) {
+    ObOpRawExpr *op_expr = NULL;
+    ObRawExpr *child_0 = NULL;
+    ObRawExpr *child_1 = NULL;
+    if (OB_FALSE_IT(op_expr = static_cast<ObOpRawExpr *>(expr))) {
```

Comment:
if (OB_FALSE_IT(op_expr = static_cast<ObOpRawExpr *>(expr))) { is unnecessary. just set op_expr when define it.